File: OrientacaoObjetosChatGpt/model/ContaBancaria.java
package model;

public class ContaBancaria {

    private int numero;
    private String nome;
    private double saldo;
    private double limite;

    public boolean saca(double valor) {
        if (valor <= (saldo + limite)) {
            saldo -= valor;
            return true;
        } else {
            return false;
        }
    }

    public void deposita(double valor) {
        this.saldo += valor;
        System.out.println("Depositado " + valor);
    }

    public boolean transfere(ContaBancaria destino, double valor) {
        if (valor <= (saldo + limite)) {
            saldo -= valor;
            destino.deposita(valor);
            System.out.println("Transferido " + valor + " para a conta de " + destino.getNome());
            return true;
        } else {
            return false;
        }
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public double getSaldo() {
        return saldo;
    }

    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

    public double getLimite() {
        return limite;
    }

    public void setLimite(double limite) {
        this.limite = limite;
    }

    
}

File: python/model/conta_bancaria.py
class ContaBancaria:
    def __init__(self, numero=0, nome="", saldo=0.0, limite=0.0):
        self._numero = numero
        self._nome = nome
        self._saldo = saldo
        self._limite = limite

    def saca(self, valor):
        if valor <= (self._saldo + self._limite):
            self._saldo -= valor
            return True
        else:
            return False

    def deposita(self, valor):
        self._saldo += valor
        print(f"Depositado {valor}")

    def transfere(self, destino, valor):
        if self.saca(valor):
            destino.deposita(valor)
            print(f"Transferido {valor} para a conta de {destino.nome}")
            return True
        else:
            return False

    @property
    def numero(self):
        return self._numero

    @numero.setter
    def numero(self, numero):
        self._numero = numero

    @property
    def nome(self):
        return self._nome

    @nome.setter
    def nome(self, nome):
        self._nome = nome

    @property
    def saldo(self):
        return self._saldo

    @saldo.setter
    def saldo(self, saldo):
        self._saldo = saldo

    @property
    def limite(self):
        return self._limite

    @limite.setter
    def limite(self, limite):
        self._limite = limite
